.github/workflows/sphinx-lint.yml

```
@@ -0,0 +1,30 @@
+name: Lint documentation
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "docs/**"
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  doc8:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install sphinx-lint
+      - name: Linting docs
+        run: sphinx-lint docs
```

and I would change this to:
```suggestion
        run: tools/sphinx-lint.sh
```

This ensures that we use the same arguments to the tool for both CI and local usage.